### PR TITLE
Split profile functionality from build_yaml

### PR DESCRIPTION
**Pull request by matejak**
No functional changes at all - just move the code from one big file to separate ones. Import the symbols to the original file for compatibility.

### CODE REVIEW

File: ssg/build_yaml.py [hunk 1]
```diff
@@ -51,53 +46,16 @@
...
-    for x in element.findall(".//{%s}sub" % XCCDF12_NS):
-        x.set("idref", OSCAP_VALUE + x.get("idref"))
-        x.set("use", "legacy")
-    parent.append(element)
-    return element
+from .entities.common import (
+    XCCDFEntity,
+    SelectionHandler,
+    add_sub_element,
+    derive_id_from_file_name
```
Comment (yuumasato) [inline ssg/build_yaml.py:53]: `SelectionHandler` and `derive_id_from_file_name` don't seem to be used anywhere in this file.

File: ssg/build_yaml.py [hunk 2]
```diff
@@ -51,53 +46,16 @@
...
+)
+from .entities.profile import (
+    Profile,
+    ProfileWithInlinePolicies,
```
Comment (yuumasato) [inline ssg/build_yaml.py:57]: Only `ProfileWithInlinePolicies` is used in this file.

**Review by yuumasato — COMMENTED** [review]
Is there a reason not to unify `profile_base.py` and `profile.py`?

**Comment by matejak** [comment]
> Is there a reason not to unify `profile_base.py` and `profile.py`?

Basically just to keep the file size and the imports section small / focused.

File: ssg/build_yaml.py (hunk 1 above)
Comment (matejak) [reply ssg/build_yaml.py:53]: I see, no reason to maintain compatibility that nobody is interested in.

File: ssg/build_yaml.py (hunk 2 above)
Comment (matejak) [reply ssg/build_yaml.py]: Profile is used by other scripts (generating tables), so I would keep it here, as `build_yaml` could stay a "hub", where things are available for imports, but are defined elsewhere.

**Review by jan-cerny — CHANGES_REQUESTED** [review]
I like the idea very much but please resolve the conlfict

**Comment by matejak** [comment]
As the PR just moves stuff around, I suggest to disregard CodeClimate warnings, and address those later, to get this merged without delays.

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**